tests_integration/test_integration.py

```diff
@@ -0,0 +1,233 @@
+import contextlib
+import logging
+import os
+import subprocess
+import tempfile
+from pathlib import Path
+
+import pytest
+
+from tests_integration import setup_repositories
+from tests_integration.collect_test_scenarios import get_test_scenario
+from tests_integration.lib.shared import setup_logging
+from tests_integration.lib.test_case import TestCase
+from tests_integration.step_prepare import (
+    close_all_open_pull_requests,
+    reset_cf_graph,
+    run_all_prepare_functions,
+)
+from tests_integration.step_validate import run_all_validate_functions
+
+TESTS_INTEGRATION_DIR = Path(__file__).parent
+
+
+@pytest.fixture(scope="module", autouse=True)
+def global_environment_setup():
+    """
+    Set up the global environment variables for the tests.
+    If we once migrate to pydantic-settings, this should be more fine-grained for each bot step.
+    """
+    # assert os.environ.get("BOT_TOKEN"), "BOT_TOKEN must be set."
+    assert os.environ.get("MITMPROXY_PEM"), "MITMPROXY_PEM must be set."
+    assert os.environ.get("SSL_CERT_FILE"), "SSL_CERT_FILE must be set."
+    assert os.environ.get("REQUESTS_CA_BUNDLE"), "REQUESTS_CA_BUNDLE must be set."
+    assert os.environ.get("TEST_SETUP_TOKEN"), "TEST_SETUP_TOKEN must be set."
+
+    # os.environ["CF_TICK_OVERRIDE_CONDA_FORGE_ORG"] = "conda-forge-bot-staging"
+    # os.environ["CF_TICK_GRAPH_DATA_BACKENDS"] = "file"
+    # os.environ["CF_FEEDSTOCK_OPS_CONTAINER_NAME"] = "conda-forge-tick"
+    # os.environ["CF_FEEDSTOCK_OPS_CONTAINER_TAG"] = "test"
+    # os.environ["CF_TICK_GRAPH_GITHUB_BACKEND_REPO"] = (
+    #    "regro-staging/cf-graph-countyfair"
+    # )
+    # github_run_id = os.environ.get("GITHUB_RUN_ID", "GITHUB_RUN_ID_NOT_SET")
+    # os.environ["RUN_URL"] = (
+    #    f"https://github.com/regro/cf-scripts/actions/runs/{github_run_id}"
+    # )
+
+    # os.environ["CF_TICK_FRAC_MAKE_GRAPH"] = "1.0"
+
+    setup_logging(logging.INFO)
+
+
+@pytest.fixture(scope="module")
+def repositories_setup():
+    """
+    Set up the repositories for the tests.
+    """
+    setup_repositories.prepare_all_accounts()
+
+
+@pytest.fixture(params=[0])
+def scenario(request) -> tuple[int, dict[str, TestCase]]:
+    scenario_id: int = request.param
+    close_all_open_pull_requests()
+    reset_cf_graph()
+
+    scenario = get_test_scenario(scenario_id)
+    scenario_pretty_print = {
+        feedstock_name: test_case.__class__.__name__
+        for feedstock_name, test_case in scenario.items()
+    }
+
+    print(f"Preparing test scenario {scenario_id}...")
+    print(f"Scenario: {scenario_pretty_print}")
+
+    run_all_prepare_functions(scenario)
+
+    return scenario_id, scenario
+
+
+@pytest.fixture
+def mitmproxy(scenario: tuple[int, dict[str, TestCase]]):
+    scenario_id, _ = scenario
+    print(f"Starting mitmproxy for scenario {scenario_id}...")
+
+    proxy_process = subprocess.Popen(["./mock_proxy_start.sh"])
+    yield
+    proxy_process.kill()
```

Comment:
I like https://pytest-xprocess.readthedocs.io/en/latest/ a lot for these cases. They handle cleanups and so on. Just mentioning it in case it helps. For example, I would guard the `.kill()` bit in a try / finally block.

Reply:
Thank you! This was indeed very helpful. Implemented in 56f08791e333f0dfd48dfbda4dde8f199089467f